Adding pkg-config for ncurses and slang.

kwhat opened the pull request:
HI,

I was having some trouble building on Gentoo due to ncurses using tinfo. [(457530)](https://bugs.gentoo.org/457530)  The solution is just to add pkg-config for ncurses and I went ahead and added it for slang as well.  This looks like a similar fix to what was done for freetype support.  The only part that seems a little sketchy is Makefile.usr. Please let me know if there is anything I should address. 

Thanks.

chewi commented on configure.ac:
```diff
@@ -724,22 +724,42 @@ if test $ac_lib_slang = auto; then
 		[ac_lib_slang=no],
 		[]
 	)
```
I think you meant to remove this `AC_CHECK_LIB` call like you have done for ncurses below? It's useless anyhow because the following `PKG_CHECK_MODULES` call overwrites the result.

kwhat replied:
Yep, just forgot to remove it after I created the `PKG_CHECK_MODULES`.  Fixed.

kwhat commented:
> The way that the `PKG_CHECK_MODULES` calls are repeated and the link test is only performed in the "auto" case seems odd to me. It could probably be turned inside-out with most of the code in the `PKG_CHECK_MODULES` "found" clause. It is consistent with the rest though so fair enough. 

Thanks for taking a look! I agree, but I didn't want to change more than was required for fix and since it was like that a for the other modules I just cookie cuttered it out.  I removed the leftover `PKG_CHECK_MODULES` and updated the Gentoo PR patch as well.

kwhat commented on Makefile.usr:
```diff
@@ -302,6 +302,8 @@ SDLCFLAGS = $(shell sdl-config --cflags)
 FREETYPECFLAGS = $(shell freetype-config --cflags)
 FREETYPELIBS = $(shell freetype-config --libs)
+NCURSESCFLAGS = $(shell ncurses6-config --cflags)
+NCURSESLIBS = $(shell ncurses6-config --libs)
```
@chewi I am not sure what this file is for, but it seems like it should have some `SLANGCFLAGS` and `SLANGLIBS` definitions somewhere?  I also didn't make an attempt to add anything for Windows, Dos or any other targets outside of Linux as I am not sure what the impact maybe.  Just let me know if any adjustments are need.

chewi replied:
I think this is only used for DOS and Windows. It's merged now so hey ho.
